### PR TITLE
Update deprecated jQuery methods

### PR DESCRIPTION
**Pull request by cpfergus1**
## Summary

After the inclusion of https://github.com/solidusio/solidus/pull/4167, it was discovered that some of the extensions (specifically solidus_product_assembly) specs are failing on master with no changes to the code. It was discovered that after moving to jQuery3, some methods no longer existed including one being utilized in ajax calls for splitting shipments and image uploads.

This PR corrects these issues by updating the deprecated methods to utilize those available in jQuery3.
## Checklist

Check out our [PR guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the readme to account for my changes.

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Thanks, @cpfergus1. Do you think it would be challenging to include a regression test? It looks like our test suite was failing to detect that breaking change.

**Comment by cpfergus1** [comment]
> Thanks, @cpfergus1. Do you think it would be challenging to include a regression test? It looks like our test suite was failing to detect that breaking change.

I think my latest commit might achieve that. It did not appear that we were testing AJAX call failures so I forced it by inputting bad data. This should cover "fail" and "then"

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @cpfergus1, it looks good to me!

**Review by kennyadsl — APPROVED** [review]
Thanks Connor!

@solidusio/core-team do you think we should backport this?

**Comment by waiting-for-dev** [comment]
> @solidusio/core-team do you think we should backport this?

It makes sense.

File: backend/app/assets/javascripts/spree/backend/models/image_upload.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ Spree.Models.ImageUpload = Backbone.Model.extend({
       processData: false,  // tell jQuery not to process the data
       contentType: false,  // tell jQuery not to set contentType
       xhr: function () {
-        var xhr = $.ajaxSettings.xhr();
+        var xhr = $.ajaxSetup.xhr(); // Using default ajax builder but inputting upload settings
```
Comment (elia) [inline backend/app/assets/javascripts/spree/backend/models/image_upload.js:59]: 🙏 thanks for this!

**Review by elia — APPROVED** [review]
🙌

File: backend/app/assets/javascripts/spree/backend/models/image_upload.js (hunk 1 above)
Comment (pelargir) [reply backend/app/assets/javascripts/spree/backend/models/image_upload.js:59]: Unsure how this is working for anyone. We're getting errors about `ajaxSetup.xhr` being undefined, and I cannot find references to it in anywhere else in Solidus, or in any of our external libraries including jQuery itself.

Where is `ajaxSetup.xhr` being defined?

**Pull request merged**